### ISSUE TITLE
Crash On Start-up with Blackmagic Design installed

### ISSUE THREAD

**Issue by krinux** [opentoonz/opentoonz#138]
I recently installed Opentoonz, and it keeps crashing on startup. I open the program, and it freezes on initializing toonz environment. After waiting for about a minute the program stops responding and it closes out. I do not have a logitech webcam, nor do I run on multiple monitors. I have both a C: Drive and a secondary D: drive and have tried installing it on both with no success.  My computer is up to specs, so I'm not entirely sure what the issue is. Any ideas/help?

Here are my system specs: 
OS: Windows 10 Home
Processor: Intel Core i7-5700 HQ CPU @ 2.70 GHz
2.70 GHz
RAM: 16.0 GB
GPU: NVIDIA GeForce GTX 970M

**Comment by ohsnapitsjoel** [comment]
I get crashes as well, but not every time. It's predictable for me, and I can reproduce it.  I run a few monitors, including a tablet monitor.  Fresh install of OpenToonz: if I open the application, the window opens on my main display (an Acer monitor).  As long as I exit the application with the OpenToonz window on my main display, I can restart it without a problem and even move the window to my tablet monitor.  

If I close OpenToonz while the window is on my tablet monitor, I am not able to restart the application - it crashes while loading. 

If I delete the OpenToonz 1.0 stuff > profiles > layouts >OpenToonz.user folder, I can restart the application successfully.  The settings that were saved on exit (like which monitor the window was on) are now forgotten, obviously, and the application window again opens on my main display.  Repeat ad nauseum.  

Event Viewer says the faulting module is Qt5Gui.dll installed with OpenToonz.  

Windows 10 Home x64, AMD FX-8350, Geforce GTX 960, Acer monitor and a Yiynova MSP19U+ V5.  

@krinux - you're on a laptop (I assumed from the 970M)?  Can you plug in another monitor and see if you can reproduce this? Also, can you check Event Viewer and see if you're getting the same Qt5Gui.dll error?  Event Viewer > Windows Logs > Application

![2016-04-07 12_55_51-event viewer](https://cloud.githubusercontent.com/assets/7570180/14359653/89289a78-fcc0-11e5-8d34-110737fcbeca.png)

**Comment by ohsnapitsjoel** [comment]
Whatever the underlying issue is, it's hidden in the OpenToonz stuff > profiles > layouts> OpenToonz.user > mainwindow.ini file.  If I reset it to its default values, it'll launch again, on my main display.

**Comment by krinux** [comment]
Okay so I attempted to run Opentoonz got the regular crash. So I pulled up the event manager and this is what I got. I'm not hugely computer savvy. To be quite honest I don't know what any of this means, but here it is. So here it is. As for second monitors, I do run it on a Laptop. The only other monitor I could possible use is a smaller TV that I could hook my computer up to.


![open toonz error](https://cloud.githubusercontent.com/assets/18266777/14404572/aff74142-fe3f-11e5-983d-a86021d5e23e.png)

**Comment by ss23** [comment]
@krinux Google seems to indicate it might be related to a BlackMagic product - https://www.blackmagicdesign.com/

Do you have anything of theirs plugged in? Are you able to try removing/unplugging it?

**Comment by ohsnapitsjoel** [comment]
Nope, I'm not using any BlackMagic products. I don't even know if I have Fusion 8 installed any more. :\  For me, creating a .bat file to rewrite the mainwindow.ini file on launch of OpenToonz has solved the problem.  That's a multi-monitor USB-connected tablet monitor configuration though, so it may not be the solution to a crash for other configurations, since the error may be different / unrelated.

**Comment by krinux** [comment]
@ss23 
TL;DR: I do have Black Magic software on my computer. I also have blackmagic hardware, but it is not plugged in when I try to start Open Toonz.

Funnily enough I do. I haven't used it recently, but I do have the ATEM Software Control program (Blackmagic software) on my computer. It is used in conjunction with this big live streaming monitor/box, which channels multiples cameras into the computer so I can live stream events.

When I try to start up the Open Toonz I do not have the Box connected to my computer though. I don't know much beyond that although this does sound somewhat similar to the Logitech webcam error that was resolved. If I'm correct the issue there was that OpenToonz was looking for a video input or something of the sort and the Logitech webcam caused to crash. Maybe the blackmagic existing Video inputs are causing Opentoonz to crash? Just a shot in the dark, but any help would be great!

**Comment by EraYaN** [comment]
This is still open and relevant. When build locally and run with the debugger attached, it produces an Access violation after the Black Magic Stuff is loaded (and Unloaded)

'OpenToonz_1.1.exe' (Win32): Loaded 'C:\Program Files (x86)\Blackmagic Design\Blackmagic Desktop Video\BMDCodecMJPG64.dll'. Cannot find or open the PDB file.
'OpenToonz_1.1.exe' (Win32): Loaded 'C:\Windows\System32\msvcr120.dll'. Symbols loaded.
'OpenToonz_1.1.exe' (Win32): Unloaded 'C:\Windows\System32\msvcr120.dll'
'OpenToonz_1.1.exe' (Win32): Unloaded 'C:\Program Files (x86)\Blackmagic Design\Blackmagic Desktop Video\BMDCodecMJPG64.dll'
Exception thrown at 0x000001E784FA6D20 in OpenToonz_1.1.exe: 0xC0000005: Access violation executing location 0x000001E784FA6D20.

**Comment by turtletooth** [comment]
Can you identify where the Blackmagic codecs are being loaded?  Perhaps we don't need to load them.

**Comment by EraYaN** [comment]
This is during startup (Probably to enumerate video devices?) That or when you initiate any of the windows codec magic that is included.

**Comment by RodneyBaker** [comment]
@EraYaN 
Can you confirm that this is still an issue with the most current nightly release?

Link:  https://github.com/opentoonz/opentoonz_nightlies/releases

**Comment by EraYaN** [comment]
@RodneyBaker Give me a couple of days to verify, I'm about 10000 kilometer away from my windows machine, so it will take a bit longer. Don't think I use the same compiler version though, or even have black magic stuff installed.

**Comment by nikleotide** [comment]
The crash still exists. I installed the nightly release [https://github.com/opentoonz/opentoonz_nightlies/releases/tag/nightly_25d3db05fd57aed51d666455ac8dbc44067a4cea](url) but as soon as it opens it closes.
I have a legit windows 10 running on 4 core i7 with 12 GB RAM (it's running on a virtualbox VM hosted on an ubuntu 14.04 machine. Please someone help. I have tried everything. luvcode64.dll doesn't exist on my windows32, and I don't have anything installed on my windows. I only purchased this windows for the sake of opentoonz but now nothing.. The version I installed on ubuntu is also weirdly big! Everything is so big that I cannot see the entire workspace together!

**Comment by nikleotide** [comment]
Forgot to mention, I don't have multiple monitors (unless running an os on a virtualbox is considered as the second monitor?) and nothing logitech or blackmagic related.

**Comment by shun-iwasawa** [comment]
@nikleotide 
Did you get any warning or message popup before closing?
If not, running `OpenToonz_1.2.exe` from command prompt may (hopefully) show some message on standard output.

**Comment by nikleotide** [comment]
@shun-iwasawa 
Sorry for the late reply. No I didn't get any message. It just closes right away.
I tried the command line (same result). Here is what I see after:

`walkDirectory_: C:\OpenToonz 1.2 stuff\plugins
===== PluginLoadController::finished() =====
plugin search directory:C:\OpenToonz 1.2 stuff\plugins
walkDictionary: C:\OpenToonz 1.2 stuff\plugins [dry]`

**Comment by RodneyBaker** [comment]
There are two threads of report here.  One by @EraYaN and one by @nikleotide so it would be best to break those apart especially as the original report appears to be fully on a Windows system while the other is Windows running in a VM via Linux.

@nikleotide can you open another report for your issue?
@EraYaN Any update?

**Comment by EraYaN** [comment]
The current version crashes.

Debug output (attached VS2017 to process just after starting)
```
The thread 0x788 has exited with code 0 (0x0).
'OpenToonz_1.2.exe' (Win32): Unloaded 'C:\Windows\System32\msvcr120.dll'
'OpenToonz_1.2.exe' (Win32): Unloaded 'C:\PROGRA~1\BLACKM~1\DESKTO~1\BMDCOD~2.DLL'
The thread 0x4a44 has exited with code 0 (0x0).
Exception thrown at 0x0000028CE3106D20 in OpenToonz_1.2.exe: 0xC0000005: Access violation executing location 0x0000028CE3106D20.
```

Same issue.

**Comment by RodneyBaker** [comment]
I attempted to install Blackmagic Design's Desktop Video software but I'm not having the problem after that install.

I've been using Blackmagic's software for almost as long as I've been using Opentoonz without issue.   This primarily is Fusion and Resolve. 

Keep in mind... I've never used any of their hardware and if you have some of their hardware that might make the difference.  If you have a capture card.. that would be a likely suspect.

I've seen reports online that this has to do with video codecs.

**Comment by shun-iwasawa** [comment]
Now the same crash occurs here! The crash seems to occur when loading the BlackMagic codecs on I/O setup of AVI format. I'll investigate this. (Sorry I kept you waiting for such a long time!)
